### DISCUSSION TITLE
Need help with syntax for a PNG file size limit?

### DISCUSSION THREAD

**Discussion by voidone001** [ImageMagick/ImageMagick#6426] [Help]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

Windows 11 Pro Version 22H2

### Description

I am not very good at CLI; I apologize for that in advance. 
I am trying to batch-change an entire folder of PNG files to be limited to 40MB. For simplicity, the output path can be the same as the input path. I only need to change the size; some PNGs in the folder are over 40MB, and others are not. Let me know if that needs to be corrected. 

I hope to save this script and change the path as needed. After hours of trying this another way, I feel ImageMagick might be my only option. Thank you very much for the help in advance!

**Comment by fmw42** [comment]
There is no easy solution unless your output file is JPG.  For PNG input and output, you will have to guess at resize to smaller dimensions, try, then guess a resize again iterating until convergence. Or use pngcrush or optipng, which does iteration of compression only for you. That can reduce the file size somewhat, but not a lot as you want.  See https://imagemagick.org/Usage/formats/#png_non-im

**Comment by snibgo** [comment]
@voidone001: If you have a PNG that is, say, 80 MB, what are you prepared to change to reduce that to 40MB? PNG files are lossless. You can strip metadata, and programs like pngcrush can reduce the filesize a bit, but for a major reduction you will need to reduce the number of pixels (EDIT or the pixel depth or the number of channels).
